src/sql/workbench/contrib/executionPlan/browser/executionPlanPropertiesViewBase.ts

```diff
@@ -258,10 +264,36 @@ export abstract class ExecutionPlanPropertiesViewBase extends Disposable impleme
 		this.resizeTable();
 	}
 
+	private repopulateTable() {
+		this._tableComponent.setData(this.flattenTableData(this._tableData, -1));
+		this.resizeTable();
+	}
+
 	public updateTableColumns(columns: Slick.Column<Slick.SlickData>[]) {
 		this._tableComponent.columns = columns;
 	}
 
+	public setPropertyRowsExpanded(expand: boolean): void {
+		this.setPropertyRowsExpandedHelper(this._tableComponent.getData().getItems(), expand);
+		this.repopulateTable();
+	}
+
+	/**
+	 * Expands or collapses the rows of the properties table recursively.
+	 *
+	 * @param rows The rows to be expanded or collapsed.
+	 * @param expand Flag indicating if the rows should be expanded or collapsed.
+	 */
+	private setPropertyRowsExpandedHelper(rows: Slick.SlickData[], expand: boolean): void {
```

Comment:
Not sure a helper method is that necessary here given it's not used anywhere else and is fairly short